src/components/features/forms/NewDeliveryStep3Form.tsx

```diff
@@ -0,0 +1,104 @@
+// Да го направя както step4 с уникални goodDetailsForms.
+// Да се редактира и GoodDetailsForm
+
+import { Button } from '@mui/material'
+import { useTranslation } from 'react-i18next'
+import GoodDetailsForm from './GoodDetailsForm'
+import { UseFormReturn, useFieldArray } from 'react-hook-form'
+import { useState, useEffect } from 'react'
+import { NewDeliveryStep3FormData } from '@/schemas/newDeliverySchemas'
+import { useNewDeliveryContext } from '@/hooks/useNewDeliveryContext'
+
+export default function NewDeliveryStep3Form({
+  control,
+  formState: { errors }
+}: UseFormReturn<NewDeliveryStep3FormData>) {
+  const { t: translate } = useTranslation()
+  const { formsData } = useNewDeliveryContext()
+
+  const { remove, append } = useFieldArray({
+    control,
+    name: 'goods' //the path to the goods array in formsData object, which we need to manage dynamically.
+  })
+
+  const initialGoodType = [
+    translate('newDelivery.goodType.pallets'),
```

Comment:
dont use translations, rather have something like this
```ts
enum GoodType {
   Pallets = "pallets",
   Packages = "packages",
   Pieces = "pieces"
}

const initialGoodTypes = [
   { title: translate("..palets"), value: GoodType.pallets }
   // same for others
]
```
and you would use the value as the good type